lib/tasks/publishing_api.rake

```diff
@@ -241,9 +241,14 @@ namespace :publishing_api do
   end
 
   desc "Republish all HTML attachments to the Publishing API"
-  task republish_html_attachments: :environment do
-    HtmlAttachment.where(attachable_type: "Edition", attachable_id: Edition.publicly_visible.select("id")).each do |a|
-      Whitehall::PublishingApi.bulk_republish_async(a)
+  task :republish_html_attachments, [:state] => :environment do |_, args|
+    args.with_defaults(state: "withdrawn")
+    publications = Publications.where(state: args[:state])
+    publications.find_each do |publication|
+      if !publication.html_attachments.nil? || publication.html_attachments.count.positive?
+        document = Document.find_by(id: publication.document_id)
```

Comment:
You could do `publication.document` here. Thought I don't think you need to do this line at all and can instead change the below line to have `publication.document_id`.